portable/GCC/ARM_CRx_MPU/portmacro_asm.h

```diff
@@ -0,0 +1,515 @@
+
+/** @brief MPU_CTRL value for: Privileged Read Only, No Execute
+ *
+ * @ingroup MPU Control
+ *
+ * @note Read Only in a Privileged Operating Mode
+ * No Access in User Mode
+ * Cannot Execute Code from this region
+ */
+#define portMPU_PRIV_RO_USER_NA_NOEXEC       ( 0x1500UL )
+
+/** @brief MPU_CTRL value for: Read Only, No Execute
+ *
+ * @ingroup MPU Control
+ *
+ * @note Read Only in a Privileged Operating Mode
+ * Read Only in User Mode
+ * Cannot Execute Code from this region
+ */
+#define portMPU_PRIV_RO_USER_RO_NOEXEC       ( 0x1600UL )
+
+/** @brief MPU_CTRL value to enable an MPU Region
+ * @ingroup MPU Control
+ */
+#define portMPU_REGION_ENABLE                ( 0x01UL )
+
+/** @brief The lowest priority interrupt that is usable by the system */
+#define portLOWEST_USABLE_INTERRUPT_PRIORITY ( portLOWEST_INTERRUPT_PRIORITY - 1UL )
+
+/** This following section is used to create the proper size for the ulContext array.
+ * This array is where all registers related to a task's context are saved.
+ * The size of this array will depend on if the system is using an integrated
+ * Floating Point Unit (FPU) or not. If we are using the FPU we must save the
+ * Floating Point Status and Control Register (FPSCR),
+ * and the Floating Point Registers (FPRs). The FPSCR holds the conditional bits
+ * used for floating point calculations. The FPRs hold the actual floating point bits.
+ * The remainder of a task's context consists of the General Purpose Registers (GPRs).
+ * General Purpose Registers are used to manipulate almost all variables.
+ * The Current Program Status and Control Register, which holds the operating mode
+ * and bits that correspond to any conditional checks, such as if statements.
+ * And the Critical Nesting Depth of the task.
+ *
+ *
+ * For more information about the FPU, FPSCR, and FPRs please reference ARM's website:
+ * https://developer.arm.com/documentation/den0042/a/Floating-Point
+ *
+ * Additional information related to the Cortex R4-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0363/e/fpu-programmer-s-model
+ *
+ * Additional information related to the Cortex R5-F's FPU Implementation:
+ * https://developer.arm.com/documentation/ddi0460/d/FPU-Programmers-Model
+ *
+ * Additional information related to the ArmV7-R CPSR
+ * https://developer.arm.com/documentation/ddi0406/cb/Application-Level-Architecture/Application-Level-Programmers--Model/The-Application-Program-Status-Register--APSR-?lang=en
+ *
+ * Additional information related to the GPRs:
+ * https://developer.arm.com/documentation/ddi0406/cb/System-Level-Architecture/The-System-Level-Programmers--Model/ARM-processor-modes-and-ARM-core-registers/ARM-core-registers?lang=en
+ *
+ */
+
+/** @brief The length in ulContext for the General Purpose Registers in bytes
+ * @note There are 13 GPRs, R0-R12 each is 32 bits, so 13 registers * 4 Bytes each
+ */
+#define portGPR_LENGTH                       ( 15U * 4U )
```

Comment:
Mismatch between the #define and the comment (13 vs 15). Same below for portREGISTER_CONTEXT_LENGTH